src/System.Runtime.InteropServices.RuntimeInformation/src/OSPlatform.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSPlatform : IEquatable<OSPlatform>
+    {
+        private readonly string _osPlatformString;
+
+        private const string WindowsString = "WINDOWS";
+        private const string LinuxString = "LINUX";
+        private const string OSXString = "OSX";
+
+        public static OSPlatform Windows
+        {
+            get
+            {
+                return new OSPlatform(WindowsString);
+            }
+        }
+
+        public static OSPlatform Linux
+        {
+            get
+            {
+                return new OSPlatform(LinuxString);
+            }
+        }
+
+        public static OSPlatform OSX
+        {
+            get
+            {
+                return new OSPlatform(OSXString);
+            }
+        }
+
+        internal OSPlatform(string osPlatformString)
+        {
+            if (osPlatformString == null) throw new ArgumentNullException("name");
+            if (osPlatformString.Length == 0) throw new ArgumentException(SR.Argument_EmptyValue, "name");
```

Comment:
I kept it as osPlatformString to reiterate the idea that we just expect a string. If it's weird, can rename it to osPlatform.